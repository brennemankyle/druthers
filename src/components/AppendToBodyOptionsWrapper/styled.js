import styled from "@emotion/styled";
import { StyledAppendToBodyOptionsWrapper as RawStyledAppendToBodyOptionsWrapper } from "./AppendToBodyOptionsWrapper";

const StyledAppendToBodyOptionsWrapper = styled(
  RawStyledAppendToBodyOptionsWrapper
)`
  position: absolute;
  background-color: ${props => props.styles_colors_background};
  width: ${props => props.styles_width}px;

  ${props =>
    props.placeOptionsAbove
      ? `bottom: ${window.innerHeight - props.parentRect.y}px;
      left: ${props.parentRect.x}px;`
      : `top: ${props.parentRect.y + props.parentRect.height}px;
      left: ${props.parentRect.x}px;`}
  ${props => props.styles_rightToLeft && `transform: scaleX(-1);`}
  ${props =>
    !props.styles_hasOptions && !props.styles_multiple && `display: none;`}
`;

export { StyledAppendToBodyOptionsWrapper };
