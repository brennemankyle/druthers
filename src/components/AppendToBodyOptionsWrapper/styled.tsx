import React from "react";
import styled from "@emotion/styled";
import {
  StyledAppendToBodyOptionsWrapper as RawStyledAppendToBodyOptionsWrapper,
  StyledAppendToBodyOptionsWrapperProps,
} from "./AppendToBodyOptionsWrapper";
import { Styles } from "../../utils/SelectTypes";
import { createBaseOptionsWrapper } from "../Wrapper/styled";

const BaseOptionsWrapper = createBaseOptionsWrapper(
  RawStyledAppendToBodyOptionsWrapper
);

const StyledAppendToBodyOptionsWrapper = styled(BaseOptionsWrapper)(
  (props: Styles & StyledAppendToBodyOptionsWrapperProps) => `
  position: absolute;
  background-color: ${props.styles_colors_background};

  ${
    props.placeOptionsAbove
      ? `bottom: ${window.innerHeight - props.parentRect.y}px;
        left: ${props.parentRect.x}px;`
      : `top: ${props.parentRect.y + props.parentRect.height}px;
        left: ${props.parentRect.x}px;`
  }
  ${props.styles_rightToLeft ? `transform: scaleX(-1);` : ``}
  ${!props.styles_hasOptions && !props.styles_multiple ? `display: none;` : ``}
`
);

export { StyledAppendToBodyOptionsWrapper };
