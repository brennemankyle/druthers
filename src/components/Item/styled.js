import styled from "@emotion/styled";
import Item from "./Item";

const Selection = styled(Item)`
  border-radius: ${props => props.styles_borderRadius};
  align-items: center;
  justify-content: space-between;
  margin-top: calc(${props => props.styles_selection_margin} / 2);
  margin-bottom: calc(${props => props.styles_selection_margin} / 2);
  padding-top: ${props => props.styles_selection_paddingTop};
  padding-bottom: ${props => props.styles_selection_paddingBottom};
  padding-left: ${props => props.styles_selection_paddingLeft};
  padding-right: ${props =>
    props.removable ? "0" : props.styles_selection_paddingRight};

  & .remove:not(svg):not(path) {
    display: flex;
    border: none;
    border-radius: ${props => props.styles_borderRadius};
    margin-top: -${props => props.styles_selection_paddingTop};
    margin-bottom: -${props => props.styles_selection_paddingBottom};
    padding-top: ${props => props.styles_selection_paddingTop};
    padding-bottom: ${props => props.styles_selection_paddingBottom};
    padding-left: ${props => props.styles_selection_paddingLeft};
    padding-right: ${props => props.styles_selection_paddingRight};
    background-color: transparent;
    font-size: 1em;

    svg {
      fill: ${props => props.styles_colors_primary};
      width: ${props => props.styles_icon_width};
      height: ${props => props.styles_icon_width};
    }

    ${props =>
      props.styles_selectionHighlighted === props.item.value &&
      `background-color: ${props.styles_colors_warning};

      svg {
        fill: ${props.styles_colors_warningBold};
      }`}
  }

  ${props =>
    props.styles_multiple
      ? `display: inline-flex;
      background-color: ${props.styles_colors_secondary};
      margin-right: ${props.styles_selection_margin};`
      : `display: flex;`}
  ${props =>
    props.styles_rightToLeft &&
    `
    transform: scaleX(-1);
    flex-direction: row-reverse;
    padding-left: ${
      props.removable ? "0" : props.styles_selection_paddingRight
    };
    padding-right: ${props.styles_selection_paddingLeft};`}
`;

const Option = styled(Item)`
  padding-top: ${props => props.styles_option_paddingTop};
  padding-bottom: ${props => props.styles_option_paddingBottom};
  padding-right: ${props => props.styles_option_paddingRight};
  padding-left: ${props => props.styles_option_paddingLeft};
  &:first-of-type{
    border-top-left-radius: ${props => props.styles_borderRadius};
    border-top-right-radius: ${props => props.styles_borderRadius};
  }
  &:last-of-type{
    border-bottom-left-radius: ${props => props.styles_borderRadius};
    border-bottom-right-radius: ${props => props.styles_borderRadius};
  }
  z-index: 999;

  ${props =>
    props.styles_optionHighlighted != null &&
    props.styles_optionHighlighted === props.item.value &&
    `background-color: ${props.styles_colors_highlight};`}
  ${props =>
    props.styles_rightToLeft &&
    `transform: scaleX(-1);
    text-align: right;`}
  ${props =>
    props.item.childGroup &&
    props.item.value == null &&
    `color: ${props.styles_colors_secondary};
    font-size: .8em;`}
  ${props =>
    props.item.childGroup && props.item.value != null && `font-weight: bold;`}
  ${props =>
    props.item.group &&
    `margin-left: ${props.item.group.split(".").length * 2}em;`}
`;

export { Selection, Option };
