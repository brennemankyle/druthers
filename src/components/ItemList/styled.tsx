import React from "react";
import styled from "@emotion/styled";
import ItemList from "./ItemList";

const SelectionList = styled(ItemList)(
  (props) => `
  ${
    props.styles_multiple
      ? `display: inline;`
      : `display: inline-block;
      &:not(:empty) {
        width: 100%;
      }`
  }
  list-style-type: none;
  padding: 0;
  margin: 0;
  color: ${props.styles_colors_text};
`
);

const OptionList = styled(ItemList)(
  (props) => `
  color: ${props.styles_colors_text};
  list-style-type: none;
  padding: 0;
  margin: 0;
  margin-block-start: 0;
  margin-block-end: 0;
  border-color: ${props.styles_colors_secondary};
  border-width: ${props.styles_borderWidth};
  border-style: ${props.styles_borderStyle};
  border-radius: ${props.styles_borderRadius};

  .truncate-show, .truncate-hide {
    cursor: pointer;
  }
`
);

export { SelectionList, OptionList };
