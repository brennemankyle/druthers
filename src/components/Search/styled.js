import styled from "@emotion/styled";
import RawSearch from "./Search";

const Search = styled(RawSearch)`
  border: none;
  outline: 0;
  font-size: 1em;
  font-family: inherit;
  background-color: transparent;
  &::-ms-clear {
    display: none;
  }
  margin-top: calc(${props => props.styles_selection_margin} / 2);
  margin-bottom: calc(${props => props.styles_selection_margin} / 2);
  padding-top: ${props => props.styles_selection_paddingTop};
  padding-bottom: ${props => props.styles_selection_paddingBottom};
  padding-right: ${props => props.styles_selection_paddingRight};
  padding-left: ${props => props.styles_selection_paddingLeft};
  ${props =>
    props.styles_rightToLeft &&
    `transform: scaleX(-1);
    text-align: right;`}
  ${props =>
    props.hide &&
    `opacity: 0;
    width: 0;
    padding: 0;
    margin: 0;`}
  ${props => props.styles_disabled && `cursor: not-allowed;`}
`;

export { Search };
