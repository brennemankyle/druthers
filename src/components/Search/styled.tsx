import styled from "@emotion/styled";
import RawSearch from "./Search";

const Search = styled(RawSearch)(
  props => `
  border: none;
  outline: 0;
  font-size: 1em;
  font-family: inherit;
  background-color: transparent;
  &::-ms-clear {
    display: none;
  }
  margin-top: calc(${props.styles_selection_margin} / 2);
  margin-bottom: calc(${props.styles_selection_margin} / 2);
  padding-top: ${props.styles_selection_paddingTop};
  padding-bottom: ${props.styles_selection_paddingBottom};
  padding-right: ${props.styles_selection_paddingRight};
  padding-left: ${props.styles_selection_paddingLeft};
  ${
    props.styles_rightToLeft
      ? `transform: scaleX(-1);
        text-align: right;`
      : ``
  }
  ${
    props.hide
      ? `opacity: 0;
        width: 0;
        padding: 0;
        margin: 0;`
      : ``
  }
  ${props.styles_disabled ? `cursor: not-allowed;` : ``}
`
);

export { Search };
