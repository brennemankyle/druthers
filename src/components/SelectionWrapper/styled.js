import styled from "@emotion/styled";
import RawSelectionWrapper from "./SelectionWrapper";

const SelectionWrapper = styled(RawSelectionWrapper)`
  border: 1px solid ${props => props.styles_colors_secondary};
  border-radius: ${props => props.styles_borderRadius};
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-top: ${props => props.styles_paddingTop};
  padding-bottom: ${props => props.styles_paddingBottom};
  padding-right: ${props => props.styles_paddingRight};
  padding-left: ${props => props.styles_paddingLeft};
  & > div:first-of-type {
    flex-grow: 1;
    flex-shrink: 1;
  }
  & > * {
    flex-shrink: 0;
  }

  svg.expand {
    fill: ${props => props.styles_colors_primary};
  }

  ${props =>
    !props.styles_disabled &&
    `&:hover {
      border-color: ${props.styles_colors_primary};
    }`}

  ${props =>
    props.areOptionsOpen &&
    `svg.expand {
      transform: scaleY(-1);
    }
    border-color: ${props.styles_colors_primary};`}

  ${props =>
    props.styles_hasOptions
      ? `.divider {
        width: 1px;
        align-self: stretch;
        background-color: ${props.styles_colors_secondary};
        margin-top: calc(.2em - ${props.styles_paddingTop});
        margin-bottom: calc(.2em - ${props.styles_paddingBottom});
        margin-left: .4em;
        margin-right: .4em;
      }

      svg.expand {
        width: ${props.styles_icon_width};
        height: ${props.styles_icon_width};
        margin-top: -1px;

        ${!props.styles_disabled &&
          `&:hover {
            opacity: 0.5;
          }`}
      }`
      : `.divider {
        display: none;
      }

      .expand {
        display: none;
      }`}

  background-color: ${props =>
    props.styles_disabled
      ? props.styles_colors_disabled
      : props.styles_colors_background};
  ${props => props.styles_disabled && `cursor: not-allowed;`}
`;

export { SelectionWrapper };
