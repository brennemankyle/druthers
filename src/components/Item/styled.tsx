import React from "react";
import styled from "@emotion/styled";
import Item from "./Item";

const Selection = styled(Item)(
  (props) => `
  color: ${props.styles_colors_text}
  border-radius: ${props.styles_borderRadius};
  align-items: center;
  justify-content: space-between;
  margin-top: calc(${props.styles_selection_margin} / 2);
  margin-bottom: calc(${props.styles_selection_margin} / 2);
  padding-top: ${props.styles_selection_paddingTop};
  padding-bottom: ${props.styles_selection_paddingBottom};
  padding-left: ${props.styles_selection_paddingLeft};
  padding-right: ${props.removable ? "0" : props.styles_selection_paddingRight};

  & .remove:not(svg):not(path) {
    display: flex;
    border: none;
    border-radius: ${props.styles_borderRadius};
    margin-top: -${props.styles_selection_paddingTop};
    margin-bottom: -${props.styles_selection_paddingBottom};
    padding-top: ${props.styles_selection_paddingTop};
    padding-bottom: ${props.styles_selection_paddingBottom};
    padding-left: ${props.styles_selection_paddingLeft};
    padding-right: ${props.styles_selection_paddingRight};
    background-color: transparent;
    font-size: 1em;

    svg {
      fill: ${props.styles_colors_primary};
      width: ${props.styles_icon_width};
      height: ${props.styles_icon_width};
    }

    ${
      props.styles_selectionHighlighted === props.item.value
        ? `background-color: ${props.styles_colors_warning};

      svg {
        fill: ${props.styles_colors_warningBold};
      }`
        : ``
    }
  }

  ${
    props.styles_multiple
      ? `display: inline-flex;
        background-color: ${props.styles_colors_secondary};
        margin-right: ${props.styles_selection_margin};`
      : `display: flex;`
  }
  ${
    props.styles_rightToLeft
      ? `
        transform: scaleX(-1);
        flex-direction: row-reverse;
        padding-left: ${
          props.removable ? "0" : props.styles_selection_paddingRight
        };
        padding-right: ${props.styles_selection_paddingLeft};`
      : ``
  }
`
);

const Option = styled(Item)(
  (props) => `
  color: ${props.styles_colors_text}
  padding-top: ${props.styles_option_paddingTop};
  padding-bottom: ${props.styles_option_paddingBottom};
  padding-right: ${props.styles_option_paddingRight};
  padding-left: ${props.styles_option_paddingLeft};
  &:first-of-type{
    border-top-left-radius: ${props.styles_borderRadius};
    border-top-right-radius: ${props.styles_borderRadius};
  }
  &:last-of-type{
    border-bottom-left-radius: ${props.styles_borderRadius};
    border-bottom-right-radius: ${props.styles_borderRadius};
  }
  z-index: 999;

  ${
    props.styles_optionHighlighted != null &&
    props.styles_optionHighlighted === props.item.value
      ? `background-color: ${props.styles_colors_highlight};`
      : ``
  }
  ${
    props.styles_rightToLeft
      ? `transform: scaleX(-1);
        text-align: right;`
      : ``
  }
  ${
    props.item.childGroup && props.item.value == null
      ? `color: ${props.styles_colors_secondary};
        font-size: .8em;`
      : ``
  }
  ${
    props.item.childGroup && props.item.value != null
      ? `font-weight: bold;`
      : ``
  }
  ${
    props.item.group
      ? `margin-left: ${props.item.group.split(".").length * 2}em;`
      : ``
  }
`
);

export { Selection, Option };
