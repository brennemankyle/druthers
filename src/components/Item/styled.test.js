import React from "react";
import renderer from "react-test-renderer";
import { mockStyles, mockItem, MockSvg } from "../../mocks";
import { Selection, Option } from "./styled";

describe("Selection", () => {
  it("styles", () => {
    const wrapper = renderer.create(
      <Selection
        {...mockStyles}
        item={mockItem}
        removable={false}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("removable = true", () => {
    const wrapper = renderer.create(
      <Selection
        {...mockStyles}
        item={mockItem}
        removable={true}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should highlight removable", () => {
    let mockHighlightedItem = {
      ...mockItem,
      value: "highlight"
    };

    const wrapper = renderer.create(
      <Selection
        {...mockStyles}
        item={mockHighlightedItem}
        styles_selectionHighlighted={mockHighlightedItem.value}
        removable={true}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("styles_multiple = true", () => {
    const wrapper = renderer.create(
      <Selection
        {...mockStyles}
        item={mockItem}
        styles_multiple={true}
        removable={false}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("styles_rightToLeft = true", () => {
    const wrapper = renderer.create(
      <Selection
        {...mockStyles}
        item={mockItem}
        styles_rightToLeft={true}
        removable={false}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("styles_rightToLeft = true and removable = true", () => {
    const wrapper = renderer.create(
      <Selection
        {...mockStyles}
        item={mockItem}
        styles_rightToLeft={true}
        removable={true}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("Option", () => {
  it("styles", () => {
    const wrapper = renderer.create(
      <Option {...mockStyles} item={mockItem} svg_Remove={MockSvg} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should highlight", () => {
    let mockHighlightedItem = {
      ...mockItem,
      value: "highlight"
    };

    const wrapper = renderer.create(
      <Option
        {...mockStyles}
        item={mockHighlightedItem}
        styles_optionHighlighted={mockHighlightedItem.value}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should not highlight if empty value", () => {
    let mockHighlightedItem = {
      ...mockItem,
      value: null
    };

    const wrapper = renderer.create(
      <Option
        {...mockStyles}
        item={mockHighlightedItem}
        styles_optionHighlighted={mockHighlightedItem.value}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("styles_rightToLeft = true", () => {
    const wrapper = renderer.create(
      <Option
        {...mockStyles}
        item={mockItem}
        styles_rightToLeft={true}
        svg_Remove={MockSvg}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("has item group and parent = true", () => {
    let mockGroupItem = {
      ...mockItem,
      group: "item group",
      parent: true
    };

    const wrapper = renderer.create(
      <Option {...mockStyles} item={mockGroupItem} svg_Remove={MockSvg} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("has item group and parent = true and value = null", () => {
    let mockGroupItem = {
      ...mockItem,
      group: "item group",
      parent: true,
      value: null
    };

    const wrapper = renderer.create(
      <Option {...mockStyles} item={mockGroupItem} svg_Remove={MockSvg} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("has item group and parent = false", () => {
    let mockGroupItem = {
      ...mockItem,
      group: "item group",
      parent: false
    };

    const wrapper = renderer.create(
      <Option {...mockStyles} item={mockGroupItem} svg_Remove={MockSvg} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
