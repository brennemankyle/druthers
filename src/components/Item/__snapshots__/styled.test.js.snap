// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option has item group and childGroup = 0 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
  font-weight: bold;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Option has item group and childGroup = 0 and value = null 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
  color: #C3C3C3;
  font-size: .8em;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val={null}
>
  Meaning of life
</li>
`;

exports[`Option has item group and childGroup = 0.0.0 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
  font-weight: bold;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Option has item group and no childGroup 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
  margin-left: 2em;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Option should highlight 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
  background-color: lightblue;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val="highlight"
>
  Meaning of life
</li>
`;

exports[`Option should not highlight if empty value 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val={null}
>
  Meaning of life
</li>
`;

exports[`Option styles 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Option styles_rightToLeft = true 1`] = `
.emotion-0 {
  padding-top: .25em;
  padding-bottom: .25em;
  padding-right: .25em;
  padding-left: .25em;
  z-index: 999;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  text-align: right;
}

.emotion-0:first-of-type {
  border-top-left-radius: .2em;
  border-top-right-radius: .2em;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: .2em;
  border-bottom-right-radius: .2em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Selection removable = true 1`] = `
.emotion-0 {
  border-radius: .2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: calc(.2em / 2);
  margin-bottom: calc(.2em / 2);
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .remove:not(svg):not(path) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  border-radius: .2em;
  margin-top: -.2em;
  margin-bottom: -.2em;
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  background-color: transparent;
  font-size: 1em;
}

.emotion-0 .remove:not(svg):not(path) svg {
  fill: black;
  width: 1em;
  height: 1em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
  <button
    className="remove"
    tabIndex="-1"
    type="button"
    val="42"
  >
    <svg>
      <path
        d="M0 0L5 5Z"
      />
    </svg>
  </button>
</li>
`;

exports[`Selection should highlight removable 1`] = `
.emotion-0 {
  border-radius: .2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: calc(.2em / 2);
  margin-bottom: calc(.2em / 2);
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .remove:not(svg):not(path) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  border-radius: .2em;
  margin-top: -.2em;
  margin-bottom: -.2em;
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  background-color: transparent;
  font-size: 1em;
  background-color: #FABAAC;
}

.emotion-0 .remove:not(svg):not(path) svg {
  fill: black;
  width: 1em;
  height: 1em;
}

.emotion-0 .remove:not(svg):not(path) svg {
  fill: #FA2222;
}

<li
  className="emotion-0"
  val="highlight"
>
  Meaning of life
  <button
    className="remove"
    tabIndex="-1"
    type="button"
    val="highlight"
  >
    <svg>
      <path
        d="M0 0L5 5Z"
      />
    </svg>
  </button>
</li>
`;

exports[`Selection styles 1`] = `
.emotion-0 {
  border-radius: .2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: calc(.2em / 2);
  margin-bottom: calc(.2em / 2);
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .remove:not(svg):not(path) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  border-radius: .2em;
  margin-top: -.2em;
  margin-bottom: -.2em;
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  background-color: transparent;
  font-size: 1em;
}

.emotion-0 .remove:not(svg):not(path) svg {
  fill: black;
  width: 1em;
  height: 1em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Selection styles_multiple = true 1`] = `
.emotion-0 {
  border-radius: .2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: calc(.2em / 2);
  margin-bottom: calc(.2em / 2);
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background-color: #C3C3C3;
  margin-right: .2em;
}

.emotion-0 .remove:not(svg):not(path) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  border-radius: .2em;
  margin-top: -.2em;
  margin-bottom: -.2em;
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  background-color: transparent;
  font-size: 1em;
}

.emotion-0 .remove:not(svg):not(path) svg {
  fill: black;
  width: 1em;
  height: 1em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Selection styles_rightToLeft = true 1`] = `
.emotion-0 {
  border-radius: .2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: calc(.2em / 2);
  margin-bottom: calc(.2em / 2);
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-left: .4em;
  padding-right: .4em;
}

.emotion-0 .remove:not(svg):not(path) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  border-radius: .2em;
  margin-top: -.2em;
  margin-bottom: -.2em;
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  background-color: transparent;
  font-size: 1em;
}

.emotion-0 .remove:not(svg):not(path) svg {
  fill: black;
  width: 1em;
  height: 1em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
</li>
`;

exports[`Selection styles_rightToLeft = true and removable = true 1`] = `
.emotion-0 {
  border-radius: .2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: calc(.2em / 2);
  margin-bottom: calc(.2em / 2);
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-left: 0;
  padding-right: .4em;
}

.emotion-0 .remove:not(svg):not(path) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  border-radius: .2em;
  margin-top: -.2em;
  margin-bottom: -.2em;
  padding-top: .2em;
  padding-bottom: .2em;
  padding-left: .4em;
  padding-right: .4em;
  background-color: transparent;
  font-size: 1em;
}

.emotion-0 .remove:not(svg):not(path) svg {
  fill: black;
  width: 1em;
  height: 1em;
}

<li
  className="emotion-0"
  val="42"
>
  Meaning of life
  <button
    className="remove"
    tabIndex="-1"
    type="button"
    val="42"
  >
    <svg>
      <path
        d="M0 0L5 5Z"
      />
    </svg>
  </button>
</li>
`;
